platform/ios/Integration Tests/Annotation Tests/MGLAnnotationViewIntegrationTests.mm

```diff
@@ -500,6 +511,248 @@ - (void)testUserLocationWithOffsetAnchorPoint {
+    
+    srand48(0);
+    for (NSInteger iter = 0; iter < 10; iter++ ) {
+        
+        double zoom = (double)(drand48()*14.0);
+        
+        NSString *activityTitle = [NSString stringWithFormat:@"Zoom to %0.1f", zoom];
+        [XCTContext runActivityNamed:activityTitle
+                               block:^(id<XCTActivity>  _Nonnull activity) {
+                                   MGLMapCamera *toCamera = [self.mapView cameraByZoomingToZoomLevel:zoom aroundAnchorPoint:anchor];
+                                   [self internalTestSelecting:annotation withCamera:toCamera];
+                               }];
+    }
+}
+
+- (void)testSelectingAnnotationWhenMapIsScaledAndRotated {
+    
+    CLLocationCoordinate2D coordinates[] = {
+        { 0.005, 0.005 },
+        { NAN, NAN }
+    };
+    
+    NSArray *annotations = [self internalAddAnnotationsAtCoordinates:coordinates];
+    MGLPointAnnotation *annotation = annotations.firstObject;
+    
+    CGPoint anchor = CGPointMake(CGRectGetMidX(self.mapView.bounds), CGRectGetMidY(self.mapView.bounds));
+    
+    srand48(0);
+    for (NSInteger iter = 0; iter < 10; iter++ ) {
+        
+        double zoom = (double)(7.0 + drand48()*7.0);
+        CLLocationDirection angle = (CLLocationDirection)((drand48()*1080.0) - 540.0);
+        
+        NSString *activityTitle = [NSString stringWithFormat:@"Zoom to %0.1f", zoom];
+        [XCTContext runActivityNamed:activityTitle
+                               block:^(id<XCTActivity>  _Nonnull activity)
+         {
+             mbgl::CameraOptions currentCameraOptions;
+             
+             currentCameraOptions.bearing = angle;
+             currentCameraOptions.anchor = mbgl::ScreenCoordinate { anchor.x, anchor.y };
+             currentCameraOptions.zoom = zoom;
+             MGLMapCamera *toCamera = [self.mapView cameraForCameraOptions:currentCameraOptions];
+             
+             [self internalTestSelecting:annotation withCamera:toCamera];
+         }];
+    }
+}
+
+
+- (void)testShowingAnnotationsThenSelectingAnimated {
+    [self internalTestShowingAnnotationsThenSelectingAnimated:YES];
+}
+
+- (void)testShowingAnnotationsThenSelecting {
+    [self internalTestShowingAnnotationsThenSelectingAnimated:NO];
+}
+
+- (void)internalTestShowingAnnotationsThenSelectingAnimated:(BOOL)animated {
+    CLLocationCoordinate2D coordinates[21];
+    srand48(0);
+    
+    for (int i = 0; i < (int)(sizeof(coordinates)/sizeof(coordinates[0])); i++)
+    {
+        coordinates[i].latitude = drand48();
+        coordinates[i].longitude = drand48();
+    }
+    coordinates[20] = CLLocationCoordinate2DMake(NAN, NAN);
+
+    NSArray *annotations = [self internalAddAnnotationsAtCoordinates:coordinates];
+
+    XCTestExpectation *showCompleted = [self expectationWithDescription:@"showCompleted"];
+
+    CGSize size = self.mapView.frame.size;
+    
+    CGFloat maximumPadding = 100;
```

Comment:
Let's make insets asymmetric (e.g. use just top and left) and larger than 100:
e.g. UIEdgeInsetsMake(size.height / 2, size.width / 3, 0, 0);

Asymmetric because we want the content center to be away from screen center.

Reply:
done.